src/System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs

```diff
@@ -380,11 +383,17 @@ public override IDirectoryInfo Root
 
         private MockFileData GetMockFileDataForRead()
         {
-            return mockFileDataAccessor.GetFile(directoryPath) ?? MockFileData.NullObject;
+            if (refreshOnNextRead)
```

Comment:
In a multithreaded scenario, is this thread-safe? I think it probably would be, but it might not be as deterministic as the file system. Now that I've said that, the file system is often not too deterministic, so it's probably okay.

Reply:
Good hint! I'd say it's fine - worst case we'd refresh more than once.